src/repr/src/row/encoding.rs

```diff
@@ -272,13 +266,22 @@ impl TryFrom<&ProtoRow> for Row {
     }
 }
 
-impl TryFrom<ProtoRow> for Row {
-    type Error = TryFromProtoError;
+impl RustType<ProtoRow> for Row {
+    fn into_proto(self: &Self) -> ProtoRow {
+        let datums = self.iter().map(|x| x.into()).collect();
+        ProtoRow { datums }
+    }
 
-    fn try_from(x: ProtoRow) -> Result<Self, Self::Error> {
-        //TODO(lluki): Revisit this when we fix #12125
-        (&x).try_into()
-            .map_err(TryFromProtoError::RowConversionError)
+    fn from_proto(proto: ProtoRow) -> Result<Self, TryFromProtoError> {
+        // TODO: Try to pre-size this.
```

Comment:
Can we have this tracked in an issue somewhere?

Reply:
I created #12631 to track this. However, the TODO originally is from @danhhz (PR #9406), so I can only speculate what information can be used to derive a better size estimate.